src/yaml/mapping.cr

```diff
@@ -96,7 +96,19 @@ module YAML
       {% end %}
     {% end %}
 
-    def initialize(%pull : ::YAML::PullParser)
+    def self.new(%pull : ::YAML::PullParser)
+      if obj = %pull.read_alias(\{{@type}})
+        return obj
+      end
+
+      instance = allocate
+      %pull.record_anchor(instance)
+
+      instance.initialize(%pull, nil)
+      instance
+    end
+
+    def initialize(%pull : ::YAML::PullParser, _dummy : Nil)
```

Comment:
What's the purpose of `_dummy` argument here?

Reply:
Just from description:

> However, right now the compiler doesn't allow both a new and an initialize with the same argument "overloadness": the initialize wins and the new is hidden. This might be fixed later. For now, to make them different overloads, I added a _dummy argument at the end.

Reply:
Fair 'nuff, could we have `TODO` or `FIXME` comment then to keep those edgy cases at bay?

Reply:
Done